Update to android v7 billing library

jamesmontemagno opened the pull request:
#631 

#### PR Classification
API change and version update.

#### PR Summary
Updated .NET SDK and various package versions, removed a property, and added error handling for network issues.
- `global.json`: Specified .NET SDK version 8.0.400.
- `InAppBilling.android.cs`: Removed `InTestingMode` property and added handling for `BillingResponseCode.NetworkError`.
- `Plugin.InAppBilling.csproj`: Updated version numbers to 9.0.0.0 and various package references.
- `InAppBillingExceptions.shared.cs`: Added `NetworkError` to `PurchaseError` enum.

jamesmontemagno committed:
Update to android v7 billing library

jamesmontemagno committed:
bump

jamesmontemagno committed:
Upgrade to latest and make net8.0 package

jamesmontemagno committed:
merge

jamesmontemagno committed:
Merge branch 'master' into androidv7billing

rezamohamed commented:
@jamesmontemagno still open question on this, should I be targeting 6.2.1 here:

    <meta-data android:name="com.google.android.play.billingclient.version" android:value="6.2.1"/>

or the latest v7 version 7.1.1 and change the android:value = "7.1.1"? https://developer.android.com/google/play/billing/release-notes

jamesmontemagno committed:
target 8 and 9

jamesmontemagno committed:
Merge branch 'master' into androidv7billing

jamesmontemagno committed:
Update onetime

jamesmontemagno committed:
Merge branch 'androidv7billing' of https://github.com/jamesmontemagno/inappbillingplugin into androidv7billing

jamesmontemagno committed:
Update Plugin.InAppBilling.csproj

jamesmontemagno committed:
Update Plugin.InAppBilling.csproj

jamesmontemagno committed:
Remove obsolete projects and update dependencies

- Removed `InAppBillingMauiTest` and `ClassLibraryTest` projects from the solution file.
- Updated `Plugin.InAppBilling.csproj` to target only net9.0 and its variants, removing older versions.
- Added `NoWarn` property to suppress warnings CA1422 and CA1416.
- Updated `Xamarin.Android.Google.BillingClient` from version `7.1.1.1` to `7.1.1.4`.
- Updated `Xamarin.AndroidX.Lifecycle.LiveData.Core` from version `2.8.7.1` to `2.8.7.4`.
- Updated `Microsoft.Maui.Essentials` for `net9.0-android` from `9.0.10` to `9.0.81` and removed it for `net8.0-android`.

jamesmontemagno committed:
Update version information in README files

Updated README.md and readme.txt to include version
support details for the In-App Purchase plugin.
Notable changes include support for .NET 9 and Android
Billing v7 in version 9, .NET 8 in version 8, and
.NET 6+ in version 7. Added references to migration
guides and requirements for targeting Android 12 or higher.
